js/customize-concurrency.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable no-extra-parens */
+
+( function( api ) {
+	'use strict';
+
+	var component;
+
+	if ( ! api.Concurrency ) {
+		api.Concurrency = {};
+	}
+
+	component = api.Concurrency;
+
+	component.data = {};
+
+	/**
+	 * Inject the functionality.
+	 *
+	 * @return {void}
+	 */
+	component.init = function() {
+		api.bind( 'ready', function() {
+			api.each( function( setting ) {
+				setting['concurrency_timestamp'] = _customizeConcurrency.session_start_timestamp;
+			} );
+
+			api.bind( 'add', function( setting ) {
+				// todo get the timestamp into the ajax response when additional settings are loaded so that we can use it here
+				setting['concurrency_timestamp'] = _customizeConcurrency.session_start_timestamp;
+			} );
+
+			api.bind( 'saved', function( data ) {
+				wp.customize.each( function( setting ) {
+					if ( data.saved_post_setting_values.hasOwnProperty( setting.ID ) ) {
```

Comment:
@lgedeon You'll not want to use `saved_post_setting_values` here because this is only relevant for `post` settings, and these settings are already concurrency-aware. You can instead rely on the `setting_validities` property to get a list of _all_ the settings that were included in the request. Then you should actually iterate over _these_ settings rather than _all_ of the settings. So you can do this instead:

```js
api.bind( 'saved', function( data ) {
	if ( ! data.concurrency_session_timestamp || ! data.setting_validities ) {
		return;
	}
	_.each( data.setting_validities, function( validity, settingId ) {
		var setting = api( settingId );
		if ( setting && true === validity ) {
			setting.concurrency_timestamp = data.concurrency_session_timestamp;
		}
	} );
} );
```